### PR TITLE
Add workflow to validate Windows AMI packer config

### PR DESCRIPTION
**Pull request by ZainRizvi**
Adds a github action with basic validation for the Windows AMI packer build

### Testing
Invalidating the config: https://github.com/pytorch/test-infra/actions/runs/9507698768/job/26207629871?pr=5335
Removing the plugin definition: https://github.com/pytorch/test-infra/actions/runs/9507718897/job/26207695962?pr=5335

### CODE REVIEW

**Review by atalman — CHANGES_REQUESTED** [review]
This does not actually validate the AMI. To validate the AMI we need to build it first, deploy to canary and then finally run whole CI and binary builds on new AMI. 

Does it validates the packer config ?

**Comment by ZainRizvi** [comment]
> This does not actually validate the AMI. To validate the AMI we need to build it first, deploy to canary and then finally run whole CI and binary builds on new AMI.
> 
> Does it validates the packer config ?

It validates the syntax of the packer config. Think of it as a linter-lite

**Review by atalman — APPROVED** [review]

File: .github/workflows/windows-ami-validation.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Windows AMI Validation
```
Comment (atalman) [inline .github/workflows/windows-ami-validation.yml:1]:
```suggestion
name: Windows AMI Packer config Validation
```

**Review by malfet — APPROVED** [review]
Sure, better than nothing, though I'm a bit surprised that one can do some validation of Windows packer script on Linux...

**Comment by ZainRizvi** [comment]
> Sure, better than nothing, though I'm a bit surprised that one can do some validation of Windows packer script on Linux...

This validation is very similar to running a terraform script validation, it's very basic, OS agnostic, and just slightly better than doing nothing. 

But it sets up a place for maybe doing better validations over time

**Pull request merged**